### DISCUSSION TITLE
How explicitly generate guids for componets when using HarvestDirectory to harvest files from a directory?

### DISCUSSION THREAD

**Discussion by EvgenPervenenko** [wixtoolset/issues#7764] [Questions] [answered]
Hello, community, have faced with problem, when harvesting files from a directory using HarvestDirectory all <Component> elements generated with Guid="*", but it is not usable for my case, if I use RemoveFolder with KeyPath="yes" for this Components, is there any way to explicitly generate guids or generate it through xslt?

**Comment by kerrywicks** [comment, marked as answer]
Assuming you have Heat set up in your project file, you could set the HarvestDirectoryGenerateGuidsNow property to true and HarvestDirectoryAutogenerateGuids to false in your property group:

```
<PropertyGroup>
		<HarvestDirectoryGenerateGuidsNow>True</HarvestDirectoryGenerateGuidsNow>
		<HarvestDirectoryAutogenerateGuids>False</HarvestDirectoryAutogenerateGuids>
</PropertyGroup>
```

May be worth checking out https://wixtoolset.org/docs/tools/heat/, which has a complete listing of properties you can set for Heat.

**Comment by EvgenPervenenko** [comment]
@kerrywicks thank you
